src/javascript/app/pages/user/set_currency.js

```diff
@@ -45,18 +45,37 @@ const SetCurrency = (() => {
                     })
                     .setVisibility(1);
             } else if (popup_action) {
-                const currencies = /multi_account|set_currency/.test(popup_action) ?
-                    getAvailableCurrencies(landing_company, payout_currencies) :
-                    getCurrencyChangeOptions(landing_company);
+                const is_virtual = Client.get('is_virtual');
+                const crypto_account = Client.hasCurrencyType('crypto');
+                let currencies = [];
+                
+                if (/multi_account|set_currency/.test(popup_action)) {
+                    if (is_virtual || crypto_account) {
+                        currencies = getVirtualAvailableCurrencies(landing_company, all_fiat);
+                    } else {
+                        currencies = getAvailableCurrencies(landing_company, payout_currencies);
+                    }
+                } else if (/switch_cryptocurrency/.test(popup_action)) {
+                    currencies = getCurrentCryptoCurrencies(all_fiat);
+                } else {
+                    currencies = getCurrencyChangeOptions(landing_company);
+                }
+
                 $('#hide_new_account').setVisibility(0);
+
+                if (currencies.length === 0){
+                    return;
+                }
+
                 $(`.show_${popup_action}`).setVisibility(1);
                 populateCurrencies(currencies);
                 onSelection($currency_list, $error, false);
 
                 const action_map = {
-                    set_currency   : localize('Set currency'),
-                    change_currency: localize('Change currency'),
-                    multi_account  : localize('Create account'),
+                    set_currency         : localize('Set currency'),
+                    change_currency      : localize('Change currency'),
+                    multi_account        : all_fiat ? localize('Create account') : localize('Add account'),
+                    switch_cryptocurrency: localize('Continue'),
```

Comment:
Wouldn't something like `switch_account` be more applicable here than `switch_cryptocurrency`?
```suggestion
                    switch_account: localize('Continue'),
```

Reply:
changed